skimage/filters/_sparse_cy.pyx

```diff
@@ -0,0 +1,25 @@
+import cython
+cimport numpy as np
+from .._shared.fused_numerics cimport np_floats
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def _correlate_sparse_offsets(np_floats[:] input, Py_ssize_t[:] indices, 
+                              Py_ssize_t[:] offsets, np_floats[:] values, 
+                              np_floats[:] output):
+    cdef Py_ssize_t i
+    cdef Py_ssize_t indices_len = indices.shape[0]
+    cdef Py_ssize_t value_len = values.shape[0]
+    cdef Py_ssize_t off
+    cdef np_floats val
+    cdef Py_ssize_t vindex
+    
+    with nogil:
+        for vindex in range(value_len):
+            off = offsets[vindex]
+            val = values[vindex]    
+            # this loop order optimises cache access, gives 10x speedup
```

Comment:
Not sure I believe that.

Reply:
LOL It was 10x on the hardware in which I first tested this. (2015 Macbook.) On more modern hardware (my X1C gen6), it's a 4x speedup, which you can see for yourself (and I think did see for yourself? 😂) live in my [SciPy 2019 skan talk](https://www.youtube.com/watch?v=0pUPNMglnaE) (see minute 20:35 onwards). I don't have any reason to believe Cython and Numba would behave differently here.

The point of the comment is to prevent people from swapping the loop to the more "intuitive" order, so I think it should be kept in *some* form ("*up to* 10x speedup"?).

Reply:
Performance is also dependent on the size of the dataset and whether it's 2D/3D (more cache misses in 3D).

Reply:
Also for anyone who might be interested: the repo for that talk: https://github.com/jni/skan-talk-scipy-2019

Reply:
Fine with me, thanks Juan!